pkg/utils/protoutils/marshal.go

```diff
@@ -98,6 +99,19 @@ func MarshalMapEmitZeroValues(from resources.Resource) (map[string]interface{},
 	return m, err
 }
 
+func MarshalMapFromProto(from proto.Message) (map[string]interface{}, error) {
+	out := &bytes.Buffer{}
+	if err := jsonpbMarshaler.Marshal(out, from); err != nil {
```

Comment:
If we are worried about backwards compatibility issues here, it may be worth setting `EnumsAsInts` on this marshaller as well. In gloo-fed we ran into a bug where sk clients couldn't read enums as strings because of how the unmarshalling worked. If we are worried about having both in cluster at the same time during upgrades, or other situations, we should most likely do this.

Reply:
Could you give more details about the problem you encountered in gloo-fed? I extensively tested this change in Gloo, so I am curious about what corner case I might have missed. I am reluctant to change these marshalling utils unless it is absolutely necessary.

Reply:
Talked to @EItanya , reporting the details of the conversation here for posterity. The issue is that if we were to use the current marshaller for the status, we would write the `status.state` enums as strings. This would work fine with the new solo kit, as we use the proto unmarshaller to deserialize the string enums into the `status` proto message. The problem arises, like Eitan mentioned above, when you consider what happens when older versions of solo-kit try to read in the new string enums: the older version let the k8s `client-go` library deserialize the status into the `status` proto using the default Go JSON unmarshaller, which would fail when the enums are represented as strings because the Go type for enums is `int32`. 

Since it is conceivable that we will have both old and new versions of solo-kit running in the same cluster, I will change the function to use a dedicated marshaller with `EnumsAsInts` set to true.